### PR TITLE
[Enhancement] Improve performance of catalog related operations

### LINKED ISSUE #12983: Improve performance of catalog related operations
## Enhancement
If a large number of database under the catalog or a large number of tables under the database, the following operation will be very slow

- show proc '/catalog/db'
- use db;

# 1. show proc '/catalog/catalog_name'
this catalog has 8000+ databases;
- now:  **Unable to return results for a long time**

image

- after: only use 1.11 sec (ADMIN SET FRONTEND CONFIG ("disable_check_db_status" = "true");)
image

# 2. use db;

this db has 25000+ tables
- now: **Unable to return results for a long time**

image

- after: set ignore_com_field_list=true;
image

### PR DESCRIPTION
**Pull request by guangxuCheng**
improve below command:
1. use catalog.db;
2. show proc '/catalog/catalog_name'
3. fix ui of '//catalog'

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12983

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|1|00.00%|[153]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|2|50.00%|[360]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ExternalDbsProcDir.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -732,6 +734,9 @@ public boolean getUseScanBlockCache() {
 
     @VarAttr(name = QUERY_CACHE_ENTRY_MAX_ROWS)
     private long queryCacheEntryMaxRows = 409600;
+
+    @VarAttr(name = IGNORE_COM_FIELD_LIST)
+    private boolean ignoreComFieldList = false;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:739]: Better to make its default value true, then the behavior won't change after upgrading.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java [hunk 2]
```diff
@@ -37,18 +39,22 @@ public ProcResult fetchResult() {
             return result;
         }
 
-        // get info
-        List<List<Comparable>> dbInfos = new ArrayList<List<Comparable>>();
+        List<List<Comparable>> dbInfos = new ArrayList<>();
+
+
+        if (!Config.disable_check_db_status) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java:45]: I think we can set a threshold, if the numbers exceed the threshold, we skip it directly. Then we don't need this configuration.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 1 above)
Comment (guangxuCheng) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:739]: The meaning of this variable is whether to skip the command. If the value of the variable is false, it means no skiping, which is as expected.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java (hunk 2 above)
Comment (guangxuCheng) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java:45]: The execution logic of different catalogs is different. Will it make users more confused? All catalogs keep the same logic, check or not check, maybe a better choice

**Review by imay — DISMISSED** [review]

**Comment by guangxuCheng** [comment]
push new pr to fix conflict

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 11 / 13 (84.62%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|1|00.00%|[153]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|2|50.00%|[360]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ExternalDbsProcDir.java|7|7|100.00%|[]|

**Review by imay — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java [hunk 3]
```diff
@@ -37,18 +39,22 @@ public ProcResult fetchResult() {
...
+            // if a large number of db under the catalog, this operation is very slow
+            dbNames = dbNames.stream().filter(dbName -> {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/ExternalDbsProcDir.java:47]: Why the operation is slow? 8000 databases is not very large. The iteration is so slow?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|1|00.00%|[153]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ExternalDbsProcDir.java|7|7|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|1|00.00%|[153]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ExternalDbsProcDir.java|7|7|100.00%|[]|

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by chaoyli** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/http/action/SystemAction.java|0|1|00.00%|[153]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/proc/ExternalDbsProcDir.java|7|7|100.00%|[]|

**Comment by chaoyli** [comment]
run starrocks_admit_test

**Comment by kateshaowanjou** [comment]
@guangxuCheng Congrats on your first PR on StarRocks！🎉

**Pull request merged**